dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/impl/NettyFrameworkImpl.java

```diff
@@ -0,0 +1,359 @@
+        chfw = bundle;
+    }
+
+    /**
+     * This is a required static reference, this won't
+     * be called until the component has been deactivated
+     *
+     * @param bundle CHFWBundle instance to unset
+     */
+    protected void unsetChfwBundle(CHFWBundle bundle) {}
+
+    @Activate
+    protected void activate(ComponentContext context, Map<String, Object> config) {
+        // use the executor service provided by Liberty
+        parentGroup = new NioEventLoopGroup(1, executorService);
+        childGroup = new NioEventLoopGroup(0, executorService);
+    }
+
+    @Deactivate
+    protected void deactivate(ComponentContext context, Map<String, Object> properties) {
+        serverStopping();
+    }
+
+    @Modified
+    protected void modified(ComponentContext context, Map<String, Object> config) {
+        if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+            Tr.event(this, tc, "Processing config", config);
+        }
+        // TODO: update any framework-specific config
+    }
+    
+    /**
+     * DS method for setting the executor service reference.
+     *
+     * @param executorService the {@link java.util.concurrent.ExecutorService} to
+     *                            queue work to.
+     */
+    @Reference(service = ExecutorService.class,
+               cardinality = ReferenceCardinality.MANDATORY)
+    protected void setExecutorService(ExecutorService executorService) {
+        this.executorService = executorService;
+    }
+
+    /**
+     * DS method for clearing the executor service reference.
+     * This is a required reference, will be called after deactivate.
+     *
+     * @param executorService the service instance to clear
+     */
+    protected void unsetExecutorService(ExecutorService executorService) {
+        this.executorService = null;
+    }
+
+    /**
+     * When notified that the server is going to stop, pre-quiesce all chains in the runtime.
+     * This will be called before services start getting torn down..
+     *
+     * @see com.ibm.wsspi.kernel.service.utils.ServerQuiesceListener#serverStopping()
+     */
+    @Override
+    public void serverStopping() {
+        if (isActive) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+                Tr.event(this, tc, "Destroying all endpoints");
+            }
+            isActive = false;
+            // If the system is configured to quiesce connections..
+            long timeout = getDefaultChainQuiesceTimeout();
+
+            // set up future to log channel stopped messages
+            for (Channel channel : activeChannels) {
+                channel.closeFuture().addListener(future -> {
+                    logChannelStopped(channel);
+                });
+            }
+            Future<?> parent = null;
+            Future<?> child = null;
+            if (parentGroup != null) {
+                parent = parentGroup.shutdownGracefully(0, timeout, TimeUnit.MILLISECONDS);
```

Comment:
Not sure, without looking at the source code if a quiet period (first param) of 0 is defaulting to 2 seconds or results in no required quiet period.

Reply:
`0` does result in no quiet period